crypto/pem/pem_lib.c

```diff
@@ -535,9 +536,8 @@ int PEM_get_EVP_CIPHER_INFO(char *header, EVP_CIPHER_INFO *cipher)
         header++;
     }
     *header = '\0';
-    cipher->cipher = enc = EVP_get_cipherbyname(p);
-    *header = c;
-    header++;
+    cipher->cipher = enc = EVP_get_cipherbyname(dekinfostart);
+    *header = separator;
```

Comment:
Look, this is way too much effort for a very small thing :)  An extra ++.  Big deal.  There are more important things to look at I'm sure :)